libbeat/processors/actions/rename_fields.go

```diff
@@ -0,0 +1,69 @@
+package actions
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type renameFields struct {
+	Fields []FieldTuple
+}
+
+type FieldTuple struct {
+	From string `config:"from"`
+	To   string `config:"to"`
+}
+
+func init() {
+	processors.RegisterPlugin("rename_fields",
+		configChecked(newRenameFields,
+			requireFields("fields")))
+}
+
+func newRenameFields(c *common.Config) (processors.Processor, error) {
+	config := struct {
+		Fields []FieldTuple `config:"fields"`
+	}{}
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("fail to unpack the rename_fields configuration: %s", err)
+	}
+
+	// TODO: Check if some fields are not allowed to be renamed
+
```

Comment:
What fields cannot be renamed?

Reply:
Not sure if we should allow to rename `@timestamp` for example.

Reply:
@ruflin are you planning to block the following two options:
```yaml
     - from: "@timestamp"
       to: "another_timestamp"
     - from: "good_timestamp"
       to: "@timstamp"
```

The last one would require time parsing?

Reply:
I was initially thinking about it but now thinking we should not add any artificial limitations here and lets see what happens.